clients/ui/frontend/src/app/pages/modelCatalog/components/globalFilters/MaxLatencyFilter.tsx

```diff
@@ -59,15 +57,34 @@ const MaxLatencyFilter: React.FC<MaxLatencyFilterProps> = ({ performanceArtifact
   // Show all available percentiles - in production this could be filtered based on backend data
   const getAvailablePercentiles = React.useCallback(() => PERCENTILE_OPTIONS, []);
 
+  // Find the currently active latency filter (if any)
+  const currentActiveFilter = React.useMemo(() => {
+    for (const fieldName of LATENCY_METRIC_FIELD_NAMES) {
+      const value = filterData[fieldName];
+      if (value !== undefined && typeof value === 'number') {
+        const parsed = parseLatencyFieldName(fieldName);
+        if (parsed) {
+          return {
+            fieldName,
+            metric: parsed.metric,
+            percentile: parsed.percentile,
+            value,
+          };
+        }
+      }
+    }
+    return null;
+  }, [filterData]);
```

Comment:
I think we can avoid creating a constant array of all the permutations of metric and percentile and keep those enums as the source of truth. I might have been overabstracting there, but you could just loop over both here:
```suggestion
  // Find the currently active latency filter (if any)
  const currentActiveFilter = React.useMemo(() => {
    for (const metric of Object.values(LatencyMetric)) {
      for (const percentile of Object.values(LatencyPercentile)) {
        const fieldName = getLatencyFieldName(metric, percentile);
        const value = filterData[fieldName];
        if (value !== undefined && typeof value === 'number') {
          return { fieldName, metric, percentile, value };
        }
      }
    }
    return null;
  }, [filterData]);
```